twindb_backup/source/mysql_source.py

```diff
@@ -188,44 +277,63 @@ def _get_backup_type(self):
 
     @property
     def type(self):
+        """Get backup copy type - full or incremental
+
+        :return: 'full' or 'incrmental'
+        :rtype: str
+        """
         return self._get_backup_type()
 
     @property
     def status(self):
+        """Backup status on a destination
+
+        :return: Backups status
+        :rtype: dict
+        """
         return self.dst.status()
 
     @property
     def parent(self):
-        full_backup = self.config.get('mysql', 'full_backup')
-        return sorted(self.status[full_backup].keys(), reverse=True)[0]
+        """
+        Get name of the parent copy.
+
+        :return: backup name of the parent copy
+        or its own name if the copy is a full copy.
+        :rtype: str
+        """
+        return sorted(self.status[self.full_backup].keys(), reverse=True)[0]
 
     @property
     def parent_lsn(self):
-        full_backup = self.config.get('mysql', 'full_backup')
-        return self.status[full_backup][self.parent]['lsn']
+        """LST of the parent backup copy.
+
+        :return: LSN of the parent or its own LSN
+        if the backup copy is a full copy.
+        """
+        return self.status[self.full_backup][self.parent]['lsn']
 
     def _parent_exists(self):
```

Comment:
It's a little bit confusing, I'll rename the function.
It does tell if an eligible full copy exists so `twindb-backup` can take incremental copy.